spec/system/viewing_brand_header_spec.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+RSpec.describe "Viewing the brand header", type: :system do
+  let(:theme) { upload_theme_component }
+
+  it "displays the brand header on mobile when `show_bar_on_mobile` setting is enabled",
+     mobile: true do
+    theme.update_setting(:show_bar_on_mobile, true)
+    theme.save!
+
+    visit("/")
+
+    expect(page).to have_css(".b-header")
+  end
+
+  it "should not display the brand header on mobile when `show_bar_on_mobile` setting is disabled",
+     mobile: true do
+    theme.update_setting(:show_bar_on_mobile, false)
+    theme.save!
+
+    visit("/")
+
+    expect(page).not_to have_css(".b-header")
+  end
```

Comment:
Out of curiosity, how much do we save if you merge these tests?

Reply:
Good point. I've fixed this.